tests/integration_tests/functional/test_feat_parity.py

```diff
@@ -0,0 +1,407 @@
+    root_disk = context.disk.copy()
+    cpu_template = context.custom["cpu_template"]
+    must_be_set = context.custom["flags_must_be_set"]
+    must_be_unset = context.custom["flags_must_be_unset"]
+
+    vm_instance = vm_builder.build(
+        kernel=context.kernel,
+        disks=[root_disk],
+        ssh_key=context.disk.ssh_key(),
+        config=context.microvm,
+        cpu_template=cpu_template,
+    )
+    vm = vm_instance.vm
+    vm.start()
+
+    cpuid = cpuid_utils.get_guest_cpuid(vm)
+    kernel_ver = get_guest_kernel_ver(vm)
+    allowed_regs = ["eax", "ebx", "ecx", "edx"]
+
+    for leaf, subleaf, reg, flags in must_be_set:
+        assert reg in allowed_regs
+        actual = cpuid[(leaf, subleaf, reg)] & flags
+        expected = flags
+        assert (
+            actual == expected
+        ), f"{cpu_template}: {kernel_ver=} {leaf=:#x} {subleaf=:#x} {reg=} {actual=:#x}, {expected=:#x}"
+
+    for leaf, subleaf, reg, flags in must_be_unset:
+        assert reg in allowed_regs
+        actual = cpuid[(leaf, subleaf, reg)] & flags
+        expected = 0
+        assert (
+            actual == expected
+        ), f"{cpu_template} {kernel_ver=} {leaf=:#x} {subleaf=:#x} {reg=} {actual=:#x}, {expected=:#x}"
+
+
+def _test_cpuid_feat_flags_matrix(
+    bin_cloner_path,
+    network_config,
+    cpu_template,
+    flags_must_be_set,
+    flags_must_be_unset,
+):
+    """
+    This launches tests matrix for CPUID feature flag checks for the given CPU template.
+    """
+
+    artifacts = ArtifactCollection(_test_images_s3_bucket())
+    # Testing matrix:
+    # - Guest kernel: Linux 4.14 & Linux 5.10
+    # - Rootfs: Ubuntu 18.04 with msr-tools package installed
+    # - Microvm: 1vCPU with 1024 MB RAM
+    microvm_artifacts = ArtifactSet(artifacts.microvms(keyword="1vcpu_1024mb"))
+    kernel_artifacts = ArtifactSet(artifacts.kernels())
+    disk_artifacts = ArtifactSet(artifacts.disks(keyword="bionic-msrtools"))
+    assert len(disk_artifacts) == 1
+
+    test_context = TestContext()
+    test_context.custom = {
+        "builder": MicrovmBuilder(bin_cloner_path),
+        "network_config": network_config,
+        "cpu_template": cpu_template,
+        "flags_must_be_set": flags_must_be_set,
+        "flags_must_be_unset": flags_must_be_unset,
+    }
+    test_matrix = TestMatrix(
+        context=test_context,
+        artifact_sets=[microvm_artifacts, kernel_artifacts, disk_artifacts],
+    )
+    test_matrix.run_test(_test_cpuid_feat_flags)
+
+
+@pytest.mark.skipif(
+    platform.machine() != "x86_64", reason="CPU templates are only available on x86_64."
+)
+@pytest.mark.skipif(
+    cpuid_utils.get_cpu_vendor()
+    not in [cpuid_utils.CpuVendor.INTEL, cpuid_utils.CpuVendor.AMD],
+    reason="CPU templates are only available on Intel and AMD.",
+)
+@pytest.mark.parametrize(
+    "cpu_template",
+    ["C3", "T2", "T2S", "T2CL"]
+    if cpuid_utils.get_cpu_vendor() == cpuid_utils.CpuVendor.INTEL
+    else ["T2A"],
+)
```

Comment:
Could we use a constant to simplify this here (and in other cases)?
```suggestion
if cpuid_utils.get_cpu_vendor() == cpuid_utils.CpuVendor.INTEL:
    SUPPORTED_TEMPLATES = ["C3", "T2", "T2S", "T2CL"]
elif cpuid_utils.get_cpu_vendor() == cpuid_utils.CpuVendor.AMD:
    SUPPORTED_TEMPLATES = ["T2A"]
else:
    SUPPORTED_TEMPLATES = []
@pytest.mark.parametrize("cpu_template", SUPPORTED_TEMPLATES )
```

Reply:
I think we'll need to use the mapping between vendor and templates in other tests as well. So how about defining this mapping in `tests/framework/utils_cpuid.py` or `tests/framework/defs.py`?

Reply:
I'm inclined to wait for https://github.com/firecracker-microvm/firecracker/pull/3326/files to get merged and then reuse the helper from utils_cpuid.py to simplify this.

Reply:
You could make a fixture that parametrizes the test right templates. Look at `firecracker_release` for an example.

Reply:
done